core/controllers/admin.py

```diff
@@ -2300,8 +2302,8 @@ def put(self) -> None:
         user_settings = user_services.get_user_settings_from_username(username)
 
         if user_settings is None:
-            raise self.InvalidInputException(
-                'User with given username does not exist.')
+            raise self.PageNotFoundException(
+                Exception('User with given username does not exist.'))
```

Comment:
Why do we need the extra Exception(), here and below?

If it's due to some issue with the PageNotFoundException class, then we might want to modify the PageNotFoundException class to behave like the others if needed ... so that we don't need the special case here and can treat it just like we treat the other errors.